frontend/gateway/grpcclient/client.go

```diff
@@ -10,19 +10,45 @@ import (
 	"time"
 
 	"github.com/gogo/googleapis/google/rpc"
+	"github.com/moby/buildkit/client/buildid"
 	"github.com/moby/buildkit/frontend/gateway/client"
 	pb "github.com/moby/buildkit/frontend/gateway/pb"
 	opspb "github.com/moby/buildkit/solver/pb"
 	"github.com/moby/buildkit/util/apicaps"
 	digest "github.com/opencontainers/go-digest"
 	specs "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/status"
 )
 
 const frontendPrefix = "BUILDKIT_FRONTEND_OPT_"
 
+func New(ctx context.Context, session, build, product string, c pb.LLBBridgeClient, w []client.WorkerInfo) (*grpcClient, error) {
+	ctx = buildid.AppendToOutgoingContext(ctx, build)
```

Comment:
I think it is better to create a wrapper around `pb.LLBBridgeClient` methods for injecting the buildid than to do it here.

Reply:
That wrapper would then be in the `client` pkg.